src/snowflake/connector/result_batch.py

```diff
@@ -36,18 +36,14 @@
 DOWNLOAD_TIMEOUT = 7  # seconds
 
 if TYPE_CHECKING:  # pragma: no cover
+    from pandas import DataFrame
+    from pyarrow import DataType, Table
```

Comment:
wouldn't this raise import error if pandas/pyarrow not installed? though it's under TYPE_CHECKING so run time might be fine, but during static checking, will this prompt error information if the two packages are not installed

Reply:
It does not cause issues. If you look at `.pre-commit-config.yaml` you'll see that we don't install either. I've disabled these modules completely in `mypy.ini`.
So these type-hints are only ever useful in devs'  actively working on the code-base. I've given this some more thought and I believe that devs should have every dependency installed instead of duplicating. Setting them to `None` failback might give a false sense to devs in their editors.